### PR TITLE
fix(sqlsmith): test execution time for pre-generated queries + increase coverage

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This is due to the observation that the runtime of pre-generated queries tests in deterministic-e2e-test suite increased significantly causing timeout.

Without any testing on runtime in place, it makes our `main` pipeline unstable. Since pre-generated queries could silently increase the runtime whenever a new snapshot is generated.

Hence we add this cap back inside `deterministic-e2e-test.sh` (7m), so that it is clear if deterministic test of pre-generated queries triggered timeout.

Before a new snapshot is registered, we also test its execution time, it should be under 6m, such that when pre-generated e2e test run, it would run at most 7min (assuming 1 min variance).

Some further tweaks for better coverage:
- Increase weights of testing joins. Additionally the point of recursion occurs in `gen_table_factor`. So we can leave `can_recurse` check out here.
- Increase weights of testing aggregations.
- Further decrease `gen_with`. Previously at `0.4` too many `with` statements were generated.
- Further, we add decay to our recursion. At p=0.3 without decay, it seems like some corner queries may be extremely complex, and take too long to run. Now we just decay it like so: `x_next = x_prev * 0.9`. With any values < 0.5 rounded to p=0.0 to stop recursion entirely.
- Increase total mv-on-mv per set: 10 -> 20.
- Set timeout per query set to 5m.
- Set timeout per query to 5s when generating, 12s for normal fuzzing.
- Set timeout for entire snapshot: 15m while generating (includes sqlsmith tests).
- Set timeout for e2e test: 7m (excludes sqlsmith tests, pretty much half the queries we run while generating).
- Setting timeout at various levels allow us to understand what's going wrong with different timeouts.
   - If query timing out -> too complex, should reduce number of query.
   - If entire query set timing out -> perhaps too many queries generated, OR too many complex queries (>3s execution time) generated.
   - If snapshot set timing is running out -> again too many queries generated, perhaps some query forms need to be throttled.
- Increase p of updates from 0.1 * 0.5 = 0.05 -> 0.2. (x4)

Here are some tweaks to ensure no TLE: 
- Decrease total queries per set: 100 -> 30. This means 30 batch and 30 stream queries generated per set.
   - This decrease is fine, partially mitigated by https://github.com/risingwavelabs/risingwave/pull/10444, since we run sqlsmith per PR now.
- Increase p of generating limit (0.2 -> 0.5).
- Decrease p of generating cross join in batch query (0.5 -> 0.1).

Fixes:
- Previously we used `parallel` task number. Which is modulo 16. So only queries 1-16 get refreshed. Now that's fixed.

Depends on https://github.com/risingwavelabs/risingwave/pull/10424. We treat queries taking too long as an error. We should not generate queries which exceed some time bound. We can limit query complexity adjusting weights or even adding a hard cap if needed.

Closes https://github.com/risingwavelabs/risingwave/issues/10360

Some other issues that won't be resolved here:
- Seems like recovery can take a while. This means we don't test any query when cluster is under recovery... but it also makes our runtime much longer because we need for recovery to complete.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
Timeout is actually due to natural joins in batch query + window table function... Maybe if I add a limit clause...?

File: src/tests/sqlsmith/scripts/gen_queries.sh [hunk 1]
```diff
@@ -207,28 +253,48 @@ check_failed_to_generate_queries() {
   fi
 }
 
+# Run it to make sure it matches our expected timing for e2e test.
+# Otherwise don't update this batch of queries yet.
+run_queries_timed() {
+  echo "" > $LOGDIR/run_deterministic.stdout.log
+  timeout "$TIME_BOUND" seq $E2E_TEST_NUM | parallel "MADSIM_TEST_SEED={} \
+    timeout 6m $MADSIM_BIN --run-sqlsmith-queries $OUTDIR/{} \
```
Comment (kwannoel) [inline src/tests/sqlsmith/scripts/gen_queries.sh:261]: Timeout check of 6m here.

**Comment by kwannoel** [comment]
This PR seems very effective. I have found 4 new bugs as a result:
- https://github.com/risingwavelabs/risingwave/issues/10496
- https://github.com/risingwavelabs/risingwave/issues/10495
- https://github.com/risingwavelabs/risingwave/issues/10494
- https://github.com/risingwavelabs/risingwave/issues/10493

**Review by lmatz — APPROVED** [review]
LGTM, Sqlsmith is helping a lot

**Pull request merged**